### PR TITLE
fix: glob and shelljs are dependencies

### PR DESCRIPTION
**Pull request by rodrigo4244**
glob and shelljs are dependencies not devDependencies

### CODE REVIEW

**Comment by gunar** [comment]
We're getting rid of wavy in favor of app-module-path so compatibility with wavy is irrelevant now.

**Comment by rodrigo4244** [comment]
Is irrelevant for us, but could be an important feature to other consumer. I think we should keep the test,  `resolveDir` and the note on `README`.

**Comment by gunar** [comment]
I see your point. Let's default to delete unused code.
If someone ever opens an issue about it, we can just merge the code back from the git history.

**Comment by rodrigo4244** [comment]
Sweet. I'm going to refactor it.

**Comment by gunar** [comment]
You don't need to refactor it. Just review my code.

**Comment by gunar** [comment]
Actually, we need to write tets to confirm this works with `app-module-path`.

**Comment by gunar** [comment]
If you use `require.resolve`, it might work with `app-module-path`. But don't worry about it.

File: index.js [hunk 1]
```diff
@@ -45,11 +38,12 @@ const makeGraceful = parser => {
   return { parse }
 }
 
-const parsers = ({ path: dirPath, graceful = true, pegOptions } = {}) => {
+const parsers = ({ path: dirPath = 'src/parsers', graceful = true, pegOptions } = {}) => {
```
Comment (gunar) [inline index.js]: remove this misleading default value

File: index.js [hunk 2]
```diff
@@ -45,11 +38,12 @@ const makeGraceful = parser => {
...
   assert.strictEqual(typeof dirPath, 'string', 'the path directory must be a string.')
   assert.strictEqual(typeof graceful, 'boolean', 'the graceful option must be a boolean.')
-  const cwd = resolveDir(dirPath)
-  const filesArray = listPegjsFiles(cwd)
+  const rulesDir = startsWith('.')(dirPath) ? path.join('src/parsers', dirPath) : dirPath
```
Comment (gunar) [inline index.js]: Get rid of this. `merge-parsers` should not be complected in any way with other stuff we build.

File: index.js [hunk 3]
```diff
@@ -45,11 +38,12 @@ const makeGraceful = parser => {
...
+  const testedDir = testPath(rulesDir)
```
Comment (gunar) [inline index.js]: You don't need to assign it to a variable. This suffices:
```js
testPath(rulesDir)
```

Maybe rename to assertValidPath.

File: README.md [hunk 4]
```diff
@@ -16,14 +16,16 @@ or
 ```js
 const mergeParsers = require('@invisible/merge-parsers')
 
-const parsers = mergeParsers({ path: './parsers' }) // path contains foo.pegjs and bar.pegjs
+const parsers = mergeParsers({ path: 'src/parsers' }) // path contains foo.pegjs and bar.pegjs
```
Comment (gunar) [inline README.md]: nooo, use relative paths!

File: README.md [hunk 5]
```diff
@@ -16,14 +16,16 @@ or
...
 
 parsers.foo() // calls the `foo` parser
 parsers.bar() // calls the `bar` parser
 ```
 
 ## Options
 
+- path - Path to parsers rules directory. Defaults to `src/parsers`.
```
Comment (gunar) [inline README.md]: Remove default

File: README.md [hunk 6]
```diff
@@ -32,12 +34,15 @@ parsers.bar() // calls the `bar` parser
 
 ## Miscellaneous information
 
-The module searches for the given path inside the current project directory and if doesn't find it, it tries to search inside `node_modules`.
-```
-first -> try to search 'path/to/your/parsers/directory'
-if not found -> try to search 'node_modules/path/to/your/parsers/directory'
+The module searches for a relative path inside `src/parsers`.
```
Comment (gunar) [inline README.md]: Nope. It should just use the relative path given, related to the file that called the function.

File: README.md [hunk 7]
```diff
@@ -32,12 +34,15 @@ parsers.bar() // calls the `bar` parser
...
+
+```js
+const mergeParsers = require('@invisible/merge-parsers')
+
+return mergeParsers({ path: './rules' })
+// the above is the same as:
+return mergeParsers({ path: 'src/parsers/rules' })
```
Comment (gunar) [inline README.md]: remove this behavior

**Comment by gunar** [comment]
You didn't rebase after review. WOOO! Thanks :-)

Looks good to me. Merging.

**Pull request merged**